core/src/display_object/graphic.rs

```diff
@@ -103,6 +142,9 @@ impl<'gc> TDisplayObject<'gc> for Graphic<'gc> {
         _instantiated_by: Instantiator,
         run_frame: bool,
     ) {
+        let shape = self.0.read().static_data.shape.clone();
+        self.0.write(context.gc_context).drawing = Drawing::from_swf_shape(&shape);
```

Comment:
Do we have to convert the `Shape` into a `Drawing`? In my mind we can keep the `Drawing` separate:

1) As soon as you run any drawing command on a `Shape.graphics`, the Shape's art is erased, even without a `graphics.clear()`, so it's not as if we have to "add" to a previous drawing. (For example, try shape.graphics.lineStyle()`).
2) `graphics.copyFrom` doesn't copy the Shape's art, only the Drawing's.
3) `graphics.readGraphicData` is the one way you can actually inspect the data of a Shape. The vast majority of content won't use it (added fairly late in Flash's life in FP 11.6) and those who do use it will mostly call it once and be done. So I think it would be better to convert the shape data to drawing commands on demand if that method is used.

Reply:
Good point. I changed this over to starting with a `Shape` and replacing it with a `Drawing` once something tries to borrow a `Shape` as a `Drawing`. This passes all the tests and the handful of movies I ran locally still seem to be fine.